### PR TITLE
Add kind e2e test for github action

### PR DESCRIPTION
**Pull request by nak3**
This patch adds kind e2e test for github action.

It is based on net-contour's https://github.com/knative-sandbox/net-contour/blob/master/.github/workflows/kind-e2e.yaml.

/cc @jmprusi @davidor @markusthoemmes

### CODE REVIEW

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+          sed 's/LoadBalancer/NodePort/g' | \
+          sed 's/imagePullPolicy:/# DISABLED: imagePullPolicy:/g' | \
+          kubectl apply -f -
+
+        # This tells the tests what namespace to look in for our kingress LB.
+        echo "GATEWAY_OVERRIDE=kourier" >> $GITHUB_ENV
+        echo "GATEWAY_NAMESPACE_OVERRIDE=kourier-system" >> $GITHUB_ENV
+
+        # Make sure all pods run in leader-elected mode.
+        kubectl -n kourier-system  scale deployment 3scale-kourier-gateway --replicas=0
```
Comment (markusthoemmes) [inline .github/workflows/kind-e2e.yaml:140]: There's nol leader election in this one I believe. We don't have to scale it down first imo.

File: .github/workflows/kind-e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    - name: Install Knative net-kourier
+      run: |
+        set -o pipefail
+
+        # Build and Publish our containers to the docker daemon (including test assets)
+        export GO111MODULE=on
+        export GOFLAGS=-mod=vendor
+        ko resolve --platform=all -Pf test/config/ -f deploy/kourier-knative.yaml | \
+          sed 's/LoadBalancer/NodePort/g' | \
+          sed 's/imagePullPolicy:/# DISABLED: imagePullPolicy:/g' | \
```
Comment (markusthoemmes) [inline .github/workflows/kind-e2e.yaml:132]: This can go now, I replaced `imagePullPolicy` with `IfNotPresent` everywhere.

**Comment by davidor** [comment]
This is very useful. Thanks :+1:

I have a question regarding the kubernetes versions. I see that this is tested with 1.17.x, 1.18.x and 1.19.x. Is there a policy in knative serving that specifies which kubernetes versions should be supported in each release?

**Comment by markusthoemmes** [comment]
@davidor This is currently under active discussion in the TOC. For now, I think aligning with serving makes the most sense, given that we'll run these tests there potentially too.

**Review by davidor — APPROVED** [review]
/lgtm

**Pull request merged**